include/coloring/pgr_edgeColoring.hpp

```diff
@@ -1,95 +1,162 @@
-                    G &graph ){
-                std::vector<pgr_vertex_color_rt> results;
-                bool bipartite = boost::is_bipartite(graph.graph);
-                if (bipartite) results = print_Bipartite(graph);
-                return results;
-            }
+//*************************************************************
+
+template < class G >
+class Pgr_edgeColoring {
+public:
+    typedef typename G::V V;
+    typedef typename G::E E;
+
+    //#new
+    typedef adjacency_list< vecS, vecS, undirectedS, no_property, size_t,
+            no_property >
+            Graph;
+
+    // typedef boost::adjacency_list < boost::listS, boost::vecS, boost::undirectedS > Graph;
+    typedef boost::graph_traits < Graph > ::edges_size_type edges_size_type;
+
+    /** @name EdgeColoring
+     * @{
+     *
+     */
+
+    /** @brief edgeColoring function
+     *
+     * It does all the processing and returns the results.
+     *
+     * @param graph      the graph containing the edges
+     *
+     * @returns results, when results are found
+     *
+     * @see [boost::edge_coloring]
+     * (https://www.boost.org/libs/graph/doc/edge_coloring.html)
+     */
+    std::vector < pgr_vertex_color_rt > edgeColoring(G &graph) {
+        std::vector < pgr_vertex_color_rt > results;
+
+        auto i_map = boost::get(boost::edge_all, graph.graph);
+
+        // vector which will store the color of all the edges in the graph
+        std::vector < edges_size_type > colors(boost::num_edges(graph.graph));
+
+        // An iterator property map which records the color of each edge
+        auto color_map = boost::make_iterator_property_map(colors.begin(), i_map);
+
+        /* abort in case of an interruption occurs (e.g. the query is being cancelled) */
+        CHECK_FOR_INTERRUPTS();
+
+        try {
+            boost::edge_coloring(graph.graph, color_map);
+        } catch (boost::exception const& ex) {
+            (void)ex;
+            throw;
+        } catch (std::exception &e) {
+            (void)e;
+            throw;
+        } catch (...) {
+            throw;
+        }
+
+        results = get_results(colors, graph);
+
+        return results;
+    }
+
+    //@}
+
+private:
+    /** @brief to get the results
+     *
+     * Uses the `colors` vector to get the results i.e. the color of every edge.
+     *
+     * @param colors      vector which contains the color of every edge
+     * @param graph       the graph containing the edges
+     *
+     * @returns `results` vector
+     */
+    std::vector < pgr_vertex_color_rt > get_results(
+        std::vector < edges_size_type > &colors,
+        const G &graph) {
+        std::vector < pgr_vertex_color_rt > results;
+
+        typename boost::graph_traits < Graph > ::edge_iterator e_i, e_end;
+
+        for (boost::tie(e_i, e_end) = edges(graph.graph); e_i != e_end; ++e_i) {
+            int64_t edge = graph[*e_i].id;
+
+            // auto src = source(*e_i, graph.graph);
```

Comment:
bad wrapping of unwanted code, use `"#if 0` "#endif`
if you want the code to run again just change the 0 to 1